tests/src/test/scala/docs/scaladsl/SerializationSpec.scala

```diff
@@ -0,0 +1,293 @@
+      .runWith(Producer.plainSink(producerSettings))
+
+    probe1.expectError() shouldBe a[org.apache.kafka.common.errors.SerializationException]
+    probe2.expectError() shouldBe a[org.apache.kafka.common.errors.SerializationException]
+    probe3.cancel()
+
+    control1.isShutdown.futureValue should be(Done)
+    control2.isShutdown.futureValue should be(Done)
+    thisStreamStaysAlive.shutdown().futureValue should be(Done)
+    consumerActor ! KafkaConsumerActor.Stop
+  }
+
+  it should "signal to sub-sources" in assertAllStagesStopped {
+    val group = createGroupId()
+    val topic = createTopic()
+
+    val (control1, partitionedProbe) =
+      Consumer
+        .plainPartitionedSource(specificRecordConsumerSettings(group), Subscriptions.topics(topic))
+        .toMat(TestSink.probe)(Keep.both)
+        .run()
+
+    partitionedProbe.request(1L)
+    val (_, subSource) = partitionedProbe.expectNext()
+    val subStream = subSource.runWith(TestSink.probe)
+
+    subStream.request(1L)
+
+    val producerSettings: ProducerSettings[String, Array[Byte]] =
+      ProducerSettings(system, new StringSerializer, new ByteArraySerializer)
+        .withBootstrapServers(bootstrapServers)
+    Source
+      .single(new ProducerRecord(topic, 0, "0", "0".getBytes(StandardCharsets.UTF_8)))
+      .runWith(Producer.plainSink(producerSettings))
+
+    subStream.expectError() shouldBe a[org.apache.kafka.common.errors.SerializationException]
+
+    control1.shutdown().futureValue should be(Done)
+  }
+
+  private def specificRecordConsumerSettings(group: String): ConsumerSettings[String, SpecificRecord] = {
+    val kafkaAvroSerDeConfig = Map[String, Any] {
+      AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG -> schemaRegistryUrl
+    }
+    val kafkaAvroDeserializer = new KafkaAvroDeserializer()
+    kafkaAvroDeserializer.configure(kafkaAvroSerDeConfig.asJava, false)
+
+    val deserializer = kafkaAvroDeserializer.asInstanceOf[Deserializer[SpecificRecord]]
+    ConsumerSettings(system, new StringDeserializer, deserializer)
+      .withBootstrapServers(bootstrapServers)
+      .withGroupId(group)
+      .withProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+  }
+}
+
+case class SampleAvroClass(var key: String, var name: String) extends SpecificRecordBase {
+
+  override def get(i: Int): AnyRef = i match {
+    case 0 => key
+    case 1 => name
+    case index => throw new AvroRuntimeException(s"Unknown index: $index")
+  }
+
+  override def put(i: Int, v: scala.Any): Unit = i match {
+    case 0 => key = asString(v)
+    case 1 => name = asString(v)
+    case index => throw new AvroRuntimeException(s"Unknown index: $index")
+  }
+
+  private def asString(v: Any) =
+    v match {
+      case utf8: Utf8 => utf8.toString
+      case _ => v.asInstanceOf[String]
+    }
+  override def getSchema: Schema = SampleAvroClass.SCHEMA$
+}
+
+object SampleAvroClass {
+  val SCHEMA$ : Schema =
```

Comment:
Why the dollar?

Reply:
That's the tradition in generated Avro classes.